### PR TITLE
Refactor as valid Origami component

### PR DESCRIPTION
**Pull request by oliverturner**
This PR brings the component inline with current Origami standards:

## Additions
- Pa11y tests and demo have been added
- Demos now use origami components
- References to SCSS mixins have been updated

## Deprecations
- Make is replaced by the `origami-build-tools` (OBT) toolchain
- `package.json`'s `scripts` block, and dev dependencies like karma, chai, mocha, etc. are replaced by OBT and `origami-ci-tools` (OCT) toolchains
- `n-ui-foundations` has been removed

## Refactorings
In order to conform to Origami's expected structure and lifecycle methods
- `nNotification` has been moved into its own module and is now exported from `main.js` with a `constructAll` method
- The `Toast` subcomponent (instances of which are the message nodes managed by Notification) has been broken out into its own module: previously named `Notification` it was easily mistaken for the primary component

### CODE REVIEW

**Comment by origamiserviceuser** [comment]
n-notification bundle size difference from 6.2.1 to 5ec5cd0c643292e360274be9ce1b31ad0a90acb7
js: 2.67kb increase (0.88kb/gzip)
An insignificant difference was also found for: css (master)

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,160 +1,27 @@
...
-
   test:
-    <<: *container_config_node8
+    docker:
+      - image: 'circleci/node:10-browsers'
     steps:
-      - *attach_workspace
-      - run:
-          name: Run tests
-          command: make test
```
Comment (edds) [inline .circleci/config.yml:98]: This no longer runs `make test`?

File: origami.json [hunk 2]
```diff
@@ -1,14 +1,43 @@
...
+	"description": "Next FT - onsite notification bar",
+	"keywords": ["notification"],
+	"brands": ["master"],
+	"origamiType": "module",
+	"origamiVersion": 1,
+	"origamiCategory": "components",
+	"support": "https://github.com/Financial-Times/n-notification/issues",
+	"supportContact": {
+		"email": "next.team@ft.com",
+		"slack": "financial-times/ft-origami"
```
Comment (edds) [inline origami.json]: Should this be `next-dev-support`?

**Review by edds — COMMENTED** [review]
The commit messages, and pull request description are missing any why for lots of these changes. It makes it really hard to review as I'm not entirely sure why the code has been moved around and the naming of classes has been changed. Or for example, why the makefile has been removed, or the circleci config completely changed.

Also given this is a large refactor will the readme need updating at all?

File: .circleci/config.yml (hunk 1 above)
Comment (oliverturner) [reply .circleci/config.yml:98]: Origami components don't use Make: `obt` runs the tests

**Comment by oliverturner** [comment]
The bulk of the changes come from a pairing session with Lee in which we made the component conform to Origami standards. I'll update the PR description with a more granular explanation of what these consist of.

File: .circleci/config.yml (hunk 1 above)
Comment (edds) [reply .circleci/config.yml:98]: Where is `obt` called?

Comment (edds) [reply .circleci/config.yml:98]: Is that the `npx origami-ci branch` line?

File: origami.json (hunk 2 above)
Comment (oliverturner) [reply origami.json]: Good idea

File: .circleci/config.yml (hunk 1 above)
Comment (oliverturner) [reply .circleci/config.yml:98]: Correct

**Review by edds — APPROVED** [review]

**Pull request merged**